src/Service/HealthCheck/HealthCheckHelper.cs

```diff
@@ -0,0 +1,268 @@
+            _httpUtility = httpUtility;
+        }
+
+        /// <summary>
+        /// GetHealthCheckResponse is the main function which fetches the HttpContext and then creates the comprehensive health check report.
+        /// Serializes the report to JSON and returns the response.  
+        /// </summary>
+        /// <param name="context">HttpContext</param>
+        /// <param name="runtimeConfig">RuntimeConfig</param>
+        /// <returns></returns>
+        public ComprehensiveHealthCheckReport GetHealthCheckResponse(HttpContext context, RuntimeConfig runtimeConfig)
+        {
+            // Create a JSON response for the comprehensive health check endpoint using the provided basic health report.
+            // If the response has already been created, it will be reused.
+            _httpUtility.ConfigureApiRoute(context);
+            LogTrace("Comprehensive Health check is enabled in the runtime configuration.");
+
+            ComprehensiveHealthCheckReport ComprehensiveHealthCheckReport = new();
+            UpdateVersionAndAppName(ref ComprehensiveHealthCheckReport);
+            UpdateDabConfigurationDetails(ref ComprehensiveHealthCheckReport, runtimeConfig);
+            UpdateHealthCheckDetails(ref ComprehensiveHealthCheckReport, runtimeConfig);
+            UpdateOverallHealthStatus(ref ComprehensiveHealthCheckReport);
+            return ComprehensiveHealthCheckReport;
+        }
+
+        // Updates the overall status by comparing all the internal HealthStatuses in the response.
+        private static void UpdateOverallHealthStatus(ref ComprehensiveHealthCheckReport comprehensiveHealthCheckReport)
+        {
+            if (comprehensiveHealthCheckReport.Checks == null)
+            {
+                comprehensiveHealthCheckReport.Status = HealthStatus.Healthy;
+                return;
+            }
+
+            comprehensiveHealthCheckReport.Status = comprehensiveHealthCheckReport.Checks?.Any(check => check.Status == HealthStatus.Unhealthy) == true
+                ? HealthStatus.Unhealthy
+                : HealthStatus.Healthy;
+        }
+
+        // Updates the AppName and Version for the Health report.
+        private static void UpdateVersionAndAppName(ref ComprehensiveHealthCheckReport response)
+        {
+            // Update the version and app name to the response.
+            response.Version = ProductInfo.GetProductVersion();
+            response.AppName = ProductInfo.GetDataApiBuilderUserAgent();
+        }
+
+        // Updates the DAB configuration details coming from RuntimeConfig for the Health report.
+        private static void UpdateDabConfigurationDetails(ref ComprehensiveHealthCheckReport ComprehensiveHealthCheckReport, RuntimeConfig runtimeConfig)
+        {
+            ComprehensiveHealthCheckReport.ConfigurationDetails = new ConfigurationDetails
+            {
+                Rest = runtimeConfig?.Runtime?.Rest != null && runtimeConfig.Runtime.Rest.Enabled,
```

Comment:
could reuse the functions IsRestEnabled IsGraphQLEnabled to determine this value